pkg/apis/istio/v1alpha2/validation/validation.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"fmt"
+	"reflect"
+
+	"istio.io/operator/pkg/apis/istio/v1alpha2"
+	"istio.io/operator/pkg/util"
+)
+
+const (
+	validationMethodName = "Validate"
+)
+
+// ValidateConfig  calls validation func for every defined element in Values
+func ValidateConfig(failOnMissingValidation bool, values *v1alpha2.Values, icpls *v1alpha2.IstioControlPlaneSpec) util.Errors {
+	var validationErrors util.Errors
+
+	validationErrors = util.AppendErrs(validationErrors, validateSubTypes(reflect.ValueOf(values).Elem(), failOnMissingValidation, values, icpls))
+
+	return validationErrors
+}
+
+func validateSubTypes(e reflect.Value, failOnMissingValidation bool, values *v1alpha2.Values, icpls *v1alpha2.IstioControlPlaneSpec) util.Errors {
+	var validationErrors util.Errors
+	var ptr reflect.Value
+	var value reflect.Value
+	var object reflect.Value
+	var finalMethod reflect.Value
```

Comment:
I don't think you need value or object - just use e.
Also, please define vars in the block that they are used, preferably near the code where they are first used, and only if := can't be used to create them, so you can do ptr := e here.

Reply:
I do need value and object as they carry different content which is used later in the code. I moved and re-arranged vars as per your request.